udata/models/extras_fields.py

```diff
@@ -20,15 +19,26 @@ def __init__(self, **kwargs):
         self.registered = {}
         super(ExtrasField, self).__init__()
 
-    def register(self, key, extra):
-        self.registered[key] = extra
+    def register(self, key, dbtype):
+        '''Register a DB type to add constraint on a given extra key'''
+        if not issubclass(dbtype, (BaseField, EmbeddedDocument)):
+            msg = 'ExtrasField can only register MongoEngine fields'
+            raise TypeError(msg)
+        self.registered[key] = dbtype
 
     def validate(self, value):
         super(ExtrasField, self).validate(value)
 
         errors = {}
         for key, value in value.items():
-            extra_cls = self.registered.get(key, DefaultExtra)
+            extra_cls = self.registered.get(key)
+
+            if not extra_cls:
+                if not isinstance(value, ALLOWED_TYPES):
```

Comment:
Maybe use https://pythonhosted.org/six/#six.string_types instead of `basestring` in `ALLOWED_TYPES` while we're at it ;-)

Reply:
Forget it, search and replace will be easy.

Reply:
Hum :thinking:
I tried to avoid using `six` until now, but maybe both `str` and `unicode`